enterprise/models.py

```diff
@@ -1241,11 +1241,12 @@ def save(self, *args, **kwargs):  # pylint: disable=arguments-differ
         Override of model save method to dynamically generate the password field and perform additional validation.
         """
         self.initialization_vector = utils.generate_aes_initialization_vector()
-        self.password = utils.encrypt_string(
-            get_random_string(length=32),
-            self.initialization_vector
-        )
-        self.full_clean()
+        if not self.pk:
+            self.password = utils.encrypt_string(
+                get_random_string(length=32),
+                self.initialization_vector
+            )
+            self.full_clean()
```

Comment:
Do you not need full_clean() for validation on updates?

Reply:
`full_clean` is a wrapper function that calls a set of other validation functions that are called by the Model admin form. I think we are calling `full_clean` here because we added the password and want to validate the model again. So, no I don't think we want to call it if we aren't changing anything inside the `save` function here.

Reply:
I think this will also create a new initialization_vector on update, but not password with that IV, so those would now be out of sync.

Reply:
I think you're changing the iv though, probably should all go inside the if

Reply:
Good catch.